File: atcoder.jp/contests/abc154/tasks/abc154_c/main.ts
import { readFileSync } from "fs";
let inputLinesIndex = 0;
const inputLines = readFileSync("/dev/stdin", "utf8").split("\n");
const input = () => inputLines[inputLinesIndex++];

const N = +input();
const A = input().split(" ").map(Number);

const ans = new Set(A).size === N;

console.log(ans ? "YES" : "NO");

File: atcoder.jp/contests/abc154/tasks/abc154_c/main.rs
use proconio::input;
use std::collections::HashSet;

fn main() {
    input! {
        n: usize,
        a: [i32; n],
    }

    let hs: HashSet<i32> = a.into_iter().collect();
    let ans = hs.len() == n;

    println!("{}", if ans { "YES" } else { "NO" });
}
